sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -0,0 +1,275 @@
+                                                            bool copyCustomProperties,
+                                                            string partitionKey) =>
+            BuildAmqpBatchFromMessages(
+                source.Select(eventData => BuildAmqpMessageFromEvent(eventData, copyCustomProperties, partitionKey)),
+                partitionKey);
+
+        /// <summary>
+        ///   Builds a batch <see cref="AmqpMessage" /> from a set of <see cref="AmqpMessage" />.
+        /// </summary>
+        ///
+        /// <param name="source">The set of messages to use as the body of the batch message.</param>
+        /// <param name="partitionKey">The partition key to annotate the AMQP message with; if no partition key is specified, the annotation will not be made.</param>
+        ///
+        /// <returns>The batch <see cref="AmqpMessage" /> containing the source messages.</returns>
+        ///
+        private static AmqpMessage BuildAmqpBatchFromMessages(IEnumerable<AmqpMessage> source,
+                                                              string partitionKey)
+        {
+            AmqpMessage batchEnvelope;
+
+            var batchMessages = source.ToList();
+
+            if (batchMessages.Count == 1)
+            {
+                batchEnvelope = batchMessages[0];
+            }
+            else
+            {
+                batchEnvelope = AmqpMessage.Create(batchMessages.Select(message =>
+                {
+                    message.Batchable = true;
+                    using var messageStream = message.ToStream();
+                    return new Data { Value = ReadStreamToArraySegment(messageStream) };
+                }));
+            }
+
+            if (!String.IsNullOrEmpty(partitionKey))
+            {
+                batchEnvelope.MessageAnnotations.Map[AmqpAnnotation.PartitionKey] = partitionKey;
+            }
+
+            batchEnvelope.Batchable = true;
+            batchEnvelope.MessageFormat = AmqpConstants.AmqpBatchedMessageFormat;
+
+            return batchEnvelope;
+        }
+
+        /// <summary>
+        ///   Builds an <see cref="AmqpMessage" /> from an <see cref="EventData" />
+        ///   source, optionally propagating the custom properties.
+        /// </summary>
+        ///
+        /// <param name="source">The event to use as the source of the message.</param>
+        /// <param name="copyCustomProperties"><c>true</c> if the custom properties of the event should be propagated to the message; otherwise, <c>false</c>.</param>
+        /// <param name="partitionKey">The partition key to annotate the AMQP message with; if no partition key is specified, the annotation will not be made.</param>
+        ///
+        /// <returns>The <see cref="AmqpMessage" /> constructed from the source event.</returns>
+        ///
+        private static AmqpMessage BuildAmqpMessageFromEvent(EventData source,
```

Comment:
is there an scenario where copyCustomProperties is "false"? Looks like it is always true.

Reply:
If I recall, there was use of it in the other half of the track one converter ... which I haven't ported to track two yet.   I'll take a note to double check once I get there (probably post-preview 2) and revisit.

Reply:
Double-checked the track one code and looked over what my plans would be.   You're right; it's a dead path.   Trimmed it out.